### DISCUSSION TITLE
How can I extract a single field from TinyDB in python?

### DISCUSSION THREAD

**Discussion by tduncklee** [msiemens/tinydb#471] [General]
I'm new to python and TinyDB and cannot find the answer to this anywhere which tells me I'm overlooking something really simple.

With the following code:

    from tinydb import TinyDB, Query
    db = TinyDB('voltage_settings_db.json')
    db.insert({'type': 'sample_times', 'times:':60})
    Setting = Query()
    result = db.search(Setting.type == 'sample_times')
    print(result)

Output: [{'type': 'sample_times', 'times:': 60}]

How do I get a single field? In this case, I need to end up with an integer of 60.

I'm looking for something like the following:

    time = result['times']
    or
    time = result.times
    or
    time = db.getfield('times')

**Comment by MrPigss** [comment]
See documentation:
> db.get(query) | Get one document matching the query

This returns a single document wich behaves just like a dictionary.

**Reply by tduncklee** [reply]
Thanks for the reply but your answer does match my question. The question was how do I get an individual **_FIELD_** from a document returned from a query. This is nowhere to be found in any of the documentation. I finally figured it out:

`time = int(result[0]['times'])
`
I realize I could have used db.get(). Then to access a field from the document it would likely be:

`time = int(result['times'])
`

**Reply by msiemens** [reply]
Yes, `db.get()` returns a single result while `db.search()` returns a list of all matching results
